### ISSUE TITLE
javascript select Identifier id tag html #

### ISSUE THREAD

**Issue by Sampaioa07** [apache/netbeans#7961]
### Apache NetBeans version

Apache NetBeans 23

### What happened

<img width="411" alt="file_js" src="https://github.com/user-attachments/assets/41cea18e-c221-4a72-bae4-e0dc16e36597">
<img width="444" alt="file_php" src="https://github.com/user-attachments/assets/31c3c361-d04b-4946-89a0-7f85b6cccc92">

Hello, there is a flaw from version 21 to 23 that in version 20, when selecting an identifier id # in the javascript file, it selects all text (name and #) 
But if it is javascript in a php file, it only selects the text, not the #.
How can I fix

### How to reproduce

Hello, there is a flaw from version 21 to 23 that in version 20, when selecting an identifier id # in the javascript file, it selects all text (name and #) 
But if it is javascript in a php file, it only selects the text, not the #.
How can I fix

### Did this work correctly in an earlier version?

No / Don't know

### Operating System

windows

### JDK

jdk-22

### Apache NetBeans packaging

Apache NetBeans provided installer

### Are you willing to submit a pull request?

Yes

**Comment by matthiasblaesing** [comment]
Do you mean, that you double clicked on the identifier and the automatic selection was different? Because manually selecting the right range will obviously work.

**Comment by Sampaioa07** [comment]
In js files it is not working double clicking on "#identifier" is always selecting the text and the # plus if it is ". identifier" only selects the text for other special characters netbens in js files only selects the text plus if it is the # select the text and #.

**Comment by Sampaioa07** [comment]
Yes, it's not an error, but I think it's an inconsistency or failure. Because from version 20 below in js files, if you double click on a text with # it will only select the text. From version 21 to 23 I stop doing this.
Only happens in js files. If the javascrip script is in a php file, double clicking on #aaaa will only select the text aaaa.
It only happens with the # of o . or other special character double clicking selects only the text not the special character.

And as incredible as it may seem, I always prefer to work on version 20...

**Comment by matthiasblaesing** [comment]
This is most probably a result of https://github.com/apache/netbeans/pull/6741. That PR adds '#' to the valid characters in identifiers in JS to support private properties `#prop`. There might be an issue hidden here, as this setting should also take hold in other environments.

**Comment by Sampaioa07** [comment]
Solved thank you very much @matthiasblaesing

**Comment by matthiasblaesing** [comment]
The problem is indeed, that the primary language is used for determining which "rules" to use for finding word boundaries. It is possible to fix this, but my gut feeling is, that it will be hard to find the right location and ensure, there are no bad side effects. Start point for the search is `BeginWordAction` in `BaseKit`. The double click is a compound, formed by the "Go to beginning of word" and "Extend selection to end of word" actions. The former needs to be fixed.

**Comment by Sampaioa07** [comment]
Hello, I tried but I can't solve it and even in the new version 24 it still has the same error.
Is it possible to put it like this in version 20 to 24

**Comment by matthiasblaesing** [comment]
@Sampaioa07 just to be clear: It is correct, that in side JS the `#` is part of the identifier. The `#` introduces private properties and thus must be recognized as part of that. What is wrong, that it is not consistent between JS files and JS sections.

**Comment by Sampaioa07** [comment]
OK. Can you recommend a solution for this?
in version 20 file js double click does not select the # you have to copy this rule to version 24

**Comment by Chris2011** [comment]
The thing here is that it happens in strings and special chars in strings like - and . should be delimiters like camelCaseNavigation also. This should be configurable IMHO for each lang. Some examples:

const foo = "#foo_bar" -> double click bar, whole string is selected
const foo2 = ".foo.bar" -> double click bar, just bar is selected, double click foo, just foo is selected
const foo3 = "foo.bar" -> same as foo2, and also double click . is selecting just .
const foo4 = "foo_bar" -> double click bar, whole string is selected
const foo5 = ".bar-baz" -> double click baz, just buz is selected
const foo6 = "fooBarBaz" -> double click Bar, whole string is selected.
const foo7 = "fooBar Baz" -> double click foo, fooBar is selected.

**Comment by Sampaioa07** [comment]
<img width="471" alt="js_netbeans_20" src="https://github.com/user-attachments/assets/3cd41c9e-4a9b-467a-adfc-1ed5a3118a86" />
<img width="470" alt="js_netbeans_24" src="https://github.com/user-attachments/assets/04be83f0-9e79-4e28-a385-da43201a38ff" />

**Comment by Chris2011** [comment]
@Sampaioa07 thx, it is still clear, what the problem is yes. I just investigate what the solution could be.

**Comment by Sampaioa07** [comment]
OK.... It is possible in the source code of version 20 to remove this rule and put it in version 24 and generate a new exc of version 24. Or I can do this.?

**Comment by Chris2011** [comment]
This thing was possible long ago under oracle, that you can create an update for a modue, but this is not a feature that comes directly as wanted, it is a side effect of this: https://github.com/apache/netbeans/pull/6741 as @matthiasblaesing alreaad mentioned. So if you have a look into the files, which is quite small change you see why this happens. And if we change this back, another stuff will break again. So this is not a revert back, it needs to be discussed what we should do with special characters in strings in JS (probably also in all other supported languages). So I guess there is no easy(?) solution for this.

**Comment by Sampaioa07** [comment]
I can't ignore "#" it because I use it a lot when passing html fields from javascript to php...
I think the solution is to continue with version 20...
Is it possible that version 25 will be resolved?

**Comment by Sampaioa07** [comment]
version 25 continues with this failure....
